packages/protocol/script/DeployERC20Airdrop.s.sol

```diff
@@ -16,39 +17,120 @@ import "../contracts/team/airdrop/ERC20Airdrop.sol";
+
+        if (block.chainid != 167_000) {
+            // If testnet, mint 50 Million tokens to the vault
+            console.log("[TEST] Minting 50 Million tokens to the airdrop contract");
+            MockERC20(bridgedTko).mint(address(airdropContract), 50_000_000_000e18);
+            console.log("[INFO] Setting Airdrop Config");
+            console.log(" - Airdrop Start Time:", airdropStartTime);
+            console.log(" - Airdrop End Time:", airdropEndTime);
+            console.log(" - Merkle Root:");
+            console.logBytes32(merkleRoot);
+
+            // Set Merkle Proof, Start Time and End Time
+            airdropContract.setConfig(uint64(airdropStartTime), uint64(airdropEndTime), merkleRoot);
+        } else {
+            // Transfer ownership to 0xf8ff2AF0DC1D5BA4811f22aCb02936A1529fd2Be
+            // This address will have ownership of the airdrop contract which will have the ability
+            // to set the airdrop start and end time, and the merkle root with
+            // setConfig(_claimStart, _claimEnd, _merkleRoot).
+            console.log(
+                "[INFO] Transferring Ownership of Airdrop Contract to 0xf8ff2AF0DC1D5BA4811f22aCb02936A1529fd2Be"
+            );
+            airdropContract.transferOwnership(0xf8ff2AF0DC1D5BA4811f22aCb02936A1529fd2Be);
+        }
+
+        // Print Deployment Addresses
+        console.log("[INFO] Deployment Addresses");
+        console.log("[INFO] Airdrop Contract:", address(airdropContract));
+        console.log("[INFO] TKO:", bridgedTko);
+    }
+
+    /// @notice Runs the deployment script to deploy the ERC20 Airdrop contract
+    /// @dev Requires the deployer private key and addresses for the vault and bridged TKO token to
+    /// be valid
+    /// The function deploys the ERC20Airdrop contract and sets up necessary approvals.
     function run() external {
+        // Start transaction broadcasting using the deployer's private key
+        vm.startBroadcast(deployerPrivKey);
+
+        // Deploys Mock TKO
+        testEnvOnly();
+
         require(deployerPrivKey != 0, "invalid deployer priv key");
-        require(vaultAddress != address(0), "invalid vault address");
-        require(bridgedTko != address(0), "invalid bridged tko address");
+        console.log("[INFO] Deploying from", vm.addr(deployerPrivKey));
 
-        vm.startBroadcast(deployerPrivKey);
+        require(bridgedTko != address(0), "invalid bridged tko address");
 
-        ERC20Airdrop(
+        // Deploy the ERC20Airdrop contract with initial parameters
+        airdropContract = ERC20Airdrop(
             deployProxy({
                 name: "ERC20Airdrop",
                 impl: address(new ERC20Airdrop()),
                 data: abi.encodeCall(
-                    ERC20Airdrop.init, (address(0), 0, 0, bytes32(0), bridgedTko, vaultAddress)
+                    ERC20Airdrop.init,
+                    (
+                        address(0), // Owner
```

Comment:
I think we need to use an owner, not `address(0)`, maybe the deployer, aka `msg.sender` here ?
@dantaik

Reply:
here we need an `owner` address , other than `address(0)`.

Reply:
nah it is good! :D, by default owner will be msg.sender!